cli/dotfiles.go

```diff
@@ -193,6 +193,18 @@ func (r *RootCmd) dotfiles() *clibase.Cmd {
 				}
 
 				_, _ = fmt.Fprintf(inv.Stdout, "Running %s...\n", script)
+
+				// Check if the script is executable and notify on error
+				scriptPath := filepath.Join(dotfilesDir, script)
+				fi, err := os.Stat(scriptPath)
+				if err != nil {
+					return xerrors.Errorf("stat %s: %w", scriptPath, err)
+				}
+
+				if fi.Mode()&0o111 == 0 {
+					return xerrors.Errorf("%s script is not executable.\nTo solve this :-\n - Clone your dotfiles repo.\n - chmod +x %s\n - git add %s and git commit -m \"Make %s executable\"\n - git push\n", script, script, script, script)
```

Comment:
I'd like to reduce the use of newlines a bit, how about this? (Also added quoting just in case.)

```suggestion
					return xerrors.Errorf("%[1]q script is not executable. You can solve this by making it executable in your dotfiles repo:\n  chmod +x %[1]q && git add %[1]q && git commit -m \"Make %[1]s executable\" && git push", script)
```

Alternatively, we could update the docs and link to it (https://coder.com/docs/v2/latest/dotfiles).